packages/backend/src/app.ts

```diff
@@ -547,7 +554,24 @@ export class App {
     // Read incoming payment
     router.get<DefaultState, SubresourceContextWithAuthenticatedStatus>(
       '/incoming-payments/:id',
-      createWalletAddressMiddleware(),
+      async (ctx, next) => {
+        const incomingPaymentService = await ctx.container.use(
+          'incomingPaymentService'
+        )
+        const incomingPayment = await incomingPaymentService.get({
+          id: ctx.params.id
+        })
+
+        if (!incomingPayment?.walletAddress) {
+          throw new OpenPaymentsServerRouteError(401, 'Unauthorized', {
+            description: 'Failed to get wallet address from incoming payment',
+            id: ctx.params.id
+          })
+        }
+
+        ctx.walletAddressUrl = incomingPayment.walletAddress.url
```

Comment:
We do not get a specific wallet address in the following OP requests (either in the query params or request body):

GET /incoming-payments/:id
GET /outgoing-payments/:id
GET /quotes/:id
POST /incoming-payments/:id/complete

We need the `walletAddressUrl` in order to pass down into `createTokenIntrospectionMiddleware`, as the middleware needs to [determine](https://github.com/interledger/rafiki/pull/2708/files#diff-efcf49ef343448125aefe358a9de22df56ac089b07acbc9d120ea35192d26ca0L98) whether the possible `identifier` on the grant matches up to the resource for the particular wallet address. 

This is why we still need to request the resource first, before we can proceed with handling the request. The status quo is returning a 401/Unauthorized response code (to not give away information about whether resource exists or not). Maybe this should be 400, but I'm not entirely sure.

Reply:
I need to add tests for this, but I would need to do some work about splitting apart these in-line middlewares into separate functions, so I can handle it in a separate PR where I do a little more refactoring.

Reply:
For the following:

GET /incoming-payments/:id
GET /outgoing-payments/:id
GET /quotes/:id

we _could_ just store these on `ctx.state` and not have to call the `<resource>Routes.get`/fetch the resource from the DB again, but a) could be done in separate PR, or b) not worth doing at all for the consistency of having the routes as they are